junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +143,97 @@ private static boolean hasParameters(Method method) {
 		}
 	}
 
+	/**
+	 * {@code DisplayNameGenerator} that generate complete sentences.
+	 *
+	 * <p>This extends the functionality of {@link ReplaceUnderscores} by
+	 * generating complete sentences display names, divided by a separator
+	 * {@link IndicativeSentencesGeneration}.
+	 *
+	 * @since 5.6
+	 */
+	@API(status = EXPERIMENTAL, since = "5.6")
+	class IndicativeSentences implements DisplayNameGenerator {
+
+		@Override
+		public String generateDisplayNameForClass(Class<?> testClass) {
+			return getDisplayNameGenerator(testClass).generateDisplayNameForClass(testClass);
+		}
+
+		@Override
+		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+			return classReplaceToIndicativeSentence(nestedClass);
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			return classReplaceToIndicativeSentence(testClass) + getSentenceSeparator(testClass)
+					+ getDisplayNameGenerator(testClass).generateDisplayNameForMethod(testClass, testMethod);
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			Class<?> enclosingParent = testClass.getEnclosingClass();
+			Optional<DisplayName> displayName = AnnotationSupport.findAnnotation(testClass, DisplayName.class);
+			Optional<DisplayNameGeneration> displayNameGeneration = AnnotationSupport.findAnnotation(testClass,
+				DisplayNameGeneration.class);
+
+			if (enclosingParent == null) {
+				return displayName.map(DisplayName::value).orElseGet(() -> generateDisplayNameForClass(testClass));
+			}
+			else {
+				if (displayNameGeneration.isPresent()) {
+					return displayName.map(DisplayName::value).orElseGet(() -> generateDisplayNameForClass(testClass));
+				}
+				else {
+					return displayName.map(name -> classReplaceToIndicativeSentence(enclosingParent)
+							+ getSentenceSeparator(testClass) + name.value()).orElseGet(
+								() -> classReplaceToIndicativeSentence(enclosingParent)
+										+ getSentenceSeparator(testClass)
+										+ getDisplayNameGenerator(testClass).generateDisplayNameForNestedClass(
+											testClass));
+				}
+			}
+		}
+
+		/**
+		 * Gets the separator for {@link IndicativeSentencesGeneration} when extracting the
+		 * annotation from {@code IndicativeSentencesGeneration}, if it doesn't find it,
+		 * then search for the parent classes, if no separator is found use @code{", "} by default.
+		 *
+		 * @param currentClass the Test Class the separator either custom or default
+		 * @return the indicative sentence separator
+		 * {@code Class.getName()}.
+		 */
+		private String getSentenceSeparator(Class<?> currentClass) {
+			Optional<IndicativeSentencesGeneration> indicativeSentencesGeneration = AnnotationSupport.findAnnotation(
+				currentClass, IndicativeSentencesGeneration.class);
+
+			if (indicativeSentencesGeneration.isPresent())
```

Comment:
Please add braces to all `if`/`else` statements.

Reply:
On it. 👍